core/spec/models/spree/base_spec.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe Spree::Base do
+  before(:all) do
+    class CreatePrefTest < ActiveRecord::Migration[4.2]
+      def self.up
+        create_table :pref_tests do |item|
+          item.string :col
+          item.text :preferences
+        end
+      end
+
+      def self.down
+        drop_table :pref_tests
+      end
+    end
+
+    @migration_verbosity = ActiveRecord::Migration[4.2].verbose
+    ActiveRecord::Migration[4.2].verbose = false
+    CreatePrefTest.migrate(:up)
+  end
+
+  after(:all) do
+    CreatePrefTest.migrate(:down)
+    ActiveRecord::Migration[4.2].verbose = @migration_verbosity
+  end
+
+  context "with a class that has a preference column but does not explicitly serialize" do
```

Comment:
I guess you omitted to test `a class that has a preference column and does explicitly serialize` because it's already covered in the specific classes specs + `spec/models/spree/preferences/preferable_spec.rb`, isn't it? 

Wondering if having these specs split over multiple files is a good thing. I mean we could move them under preferable, or move the ones in that file that depend on the methods defined in `Spree::Base` here. Thoughts?

Reply:
Hm, in `base_spec.rb` I am only testing behaviour that is usually provided by `ActiveRecord::Base.serialize`. I think this spec can go once we get rid of the `Spree::Base#preferences` in Solidus 3.

After that, the contract between a class and the `Spree::Preferences::Preferable` module is to provide a `preferences` method that returns a Hash. I do think that the classes themselves should have a test asserting that.